### DISCUSSION TITLE
Updating the Schema of a table that is DAX query to another SSAS table

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#760] [🙋♀️TE3 Q&A]
I have a co-worker with a model that has a DAX query to another model via the AAS connector, so it is an import table.
The partition for this table was edited and the DAX query adjusted.
When I try to update the schema, I get this error:
![image](https://user-images.githubusercontent.com/45298358/207672051-a071584a-d93e-4dc9-acaa-50bda5260194.png)

Is the issue here the SSO credentials I used to sign into the dataset via TE/3 do not have access to the dataset the AAS query is accessing, and would taking over the dataset then making the credential changes fix it, or is there another issue here, or is updating the schema of this type of table simply not supported?

I did try TE/3 and Analysis Services for the schema compare - same error, so that isn't it.

Wanted to get some feedback before I head down a path of taking over credentials and trying to get access to data only to find out that isn't the fix.

**Comment by otykier** [comment]
Hi Ed. I'm not 100% sure what's going on here. The error message you see originates from PBI. Since TE3 offline schema compare (i.e. without enabling "Use Analysis Services for change detection") does not support AAS as a datasource, unless using an OLEDB / ODBC provider, enabling "Use Analysis Services for change detection" will not change anything, since TE3 will use this approach regardless. So in other words, what happens when you click "update schema" is the following:

1. TE3 creates a new transaction on the PBI service.
2. TE3 then adds a new (temporary) table to the current model, and specifies an M query for this table, which will populate it with the **schema** (i.e. columns, data types) of the table that you are trying to update.
3. TE3 then instructs the PBI service to perform a refresh of the temporary table, to get it populated with data. This is the step that fails in your example. Usually, when this step fails, it indicates that PBI is not able to connect to the source, typically because of missing/incorrect credentials.
4. Once the temporary table has all the schema metadata, TE3 runs a query on the table to get a hold of the schema.
5. The transaction in step 1 is rolled back, leaving the model in the service completely untouched by this operation. But TE3 is now able to show the schema diff dialog, thanks to the info obtained in step 4.

So in other words, the credentials you use when opening the model in TE3, should not have anything to do with this issue. It all depends on whether the credentials set up on the dataset in the service are sufficient for accessing the data source.
